### PR TITLE
SHARD-1113

### PR DESCRIPTION
**Pull request by mssabr01**
Added size limits to the number of wallet addresses and number of filters allowed in eth_newFilter to prevent OOM and DOS-level performance slowdowns

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -2892,6 +2892,16 @@ export const methods = {
       countFailedResponse(api_name, 'filter not found')
       return
     }
+    if (Array.isArray(inputFilter.address) && inputFilter.address.length > 100) {
+      callback({ code: -32000, message: 'Invalid address' }, null)
+      countFailedResponse(api_name, 'Invalid address')
+      return
+    }
+    if (inputFilter.topics.length > 200) {
```
Comment (abdulazeem-tk4vr) [inline src/api.ts]: Should we not check if the topics object is of type 'array' as well?

Comment (abdulazeem-tk4vr) [reply src/api.ts]: Similar to what has been done for the address array.

**Review by aniketdivekar — DISMISSED** [review]

File: src/api.ts (hunk 1 above)
Comment (mssabr01) [reply src/api.ts]: Good call

**Comment by mssabr01** [comment]
Addressed comment, pinged Abdul to review again

**Review by abdulazeem-tk4vr — DISMISSED** [review]

**Comment by abdulazeem-tk4vr** [comment]
I see some changes were made to the package-lock as well. They were not in the scope of this task. I want to make sure that those changes are fine.

**Review by kun6fup4nd4 — DISMISSED** [review]

**Review by kun6fup4nd4 — APPROVED** [review]

**Review by abdulazeem-tk4vr — APPROVED** [review]

**Pull request merged**